lib/src/signed_video_h26x_auth.c

```diff
@@ -353,6 +446,68 @@ set_validation_status_of_items_used_in_gop_hash(h26x_nalu_list_t *nalu_list, cha
   return num_marked_items;
 }
 
+/**
+ * Verifies the integrity of the GOP hash in the video, ensuring that the data
+ * within the GOP is authentic and complete. Updates the expected and received
+ * NALU counts, and returns true if the verification is successful.
+ */
+static bool
+verify_hashes_with_sei(signed_video_t *self, int *num_expected_nalus, int *num_received_nalus)
+{
+  assert(self);
+
+  int num_expected_hashes = -1;
+  int num_received_hashes = -1;
+  char validation_status = 'P';
+  svrc_t status = SV_UNKNOWN_FAILURE;
+
+  // The verified_signature_hash indicates if the signature is verified.
+  // If the signature hash is verified, the GOP hash can be verified as well.
+  // If the signature hash is not verified, it means the SEI is corrupted, and the whole GOP status
+  // is determined by the verified_signature_hash.
+  if (self->gop_info->verified_signature_hash == 1) {
+    validation_status = '.';
+    if (!verify_gop_hash(self)) {
+      // If the signature is verified but GOP hash is not, continue validation with the
+      // hash list if it is accessible.
+      if (self->gop_info->list_idx > 0) {
+        return verify_hashes_with_hash_list(self, num_expected_nalus, num_received_nalus);
+      }
+      validation_status = 'N';
+    }
+  } else if (self->gop_info->verified_signature_hash == 0) {
+    validation_status = 'N';
+  } else {
+    // An error occurred when verifying the GOP hash. Verify without a SEI.
+    return verify_hashes_without_sei(self);
```

Comment:
If you end up here you'll not use the set `used_in_gop_hash` flags, and instead have an independent way of marking/verifying NALUs. How can you guarantee that the "correct" NALUs are verified and not a different set?